src/spec.c

```diff
@@ -1341,61 +1385,73 @@ void Indexes_Free(dict *d) {
   // cursor list is iterating through the list as well and consuming a lot of CPU
   CursorList_Empty(&RSCursors);
 
-  arrayof(IndexSpec *) specs = array_new(IndexSpec *, dictSize(d));
+  arrayof(StrongRef) specs = array_new(StrongRef, dictSize(d));
   dictIterator *iter = dictGetIterator(d);
   dictEntry *entry = NULL;
   while ((entry = dictNext(iter))) {
-    IndexSpec *sp = dictGetVal(entry);
-    specs = array_append(specs, sp);
+    StrongRef ref = dictGetRef(entry);
+    specs = array_append(specs, ref);
   }
   dictReleaseIterator(iter);
 
   for (size_t i = 0; i < array_len(specs); ++i) {
-    IndexSpec_FreeInternals(specs[i]);
+    IndexSpec_RemoveFromGlobals(specs[i]);
   }
   array_free(specs);
 }
 
+
+//---------------------------------------- atomic updates ---------------------------------------
+
+// atomic update of usage counter
+inline static void IndexSpec_IncreasCounter(IndexSpec *sp) {
+  __atomic_fetch_add(&sp->counter , 1, __ATOMIC_SEQ_CST);
```

Comment:
Why does the spec still needs counter?

Reply:
It's a statistics counter. Not related to refcounts

Reply:
Yeah, it was needed because the index might be drop and recreated and was wanted to know that on different locations. But I wonder if now its still needed...

Reply:
lets address it in another issue?